include/log.h

```diff
@@ -67,6 +67,8 @@ struct ofono_debug_desc {
 					__FILE__, __FUNCTION__ , ## arg); \
 } while (0)
 
+#define PRINTABLE_STR(s) ((s) ? (s) : "(null)")
```

Comment:
It's not everyday that you get schooled in a review...  Thanks.  

Are there any *known* cases where glibc won't do the right thing?  If not, my guess would be that ofono upstream wouldn't accept this.

Also, as a suggestion, perhaps shorten to PRINT_STR?